Argo/Types/Decoded/Applicative.swift

```diff
@@ -1,12 +1,13 @@
 /**
-  apply a `Decoded` function to a `Decoded` value
+  Conditionally apply a `Decoded` function to a `Decoded` value.
 
-  - If the function is a failing case (`.TypeMismatch`, `.MissingKey`), this will return the function's value
-  - If the function is a `.Success` case and the value is a failure case, this will return the value
-  - If both self and the function are `.Success`, the function will be applied to the unwrapped value
+  - If either the function or value arguments are `.Failure`, that `.Failure`
```

Comment:
Should there be a 3rd line here to show the precedence of which failure is returned first?